71 webhooks (#74)
livepeer/livepeer-studio-docs: docs/guides/livestreaming/webhook.mdx

---
title: Webhooks
description: Use webhooks to listen for video lifecycle events
metaTitle: Using webhooks - Livepeer Studio
metaDescription: Use webhooks to listen for video lifecycle events
---

The following includes information and instructions on how to work with Webhooks to communicate with your application:

- Use webhooks to listen for video lifecycle events
- Check webhook signatures using webhook signatures to ensure incoming webhook requests are coming from Livepeer

## What you need to Know

- Familiarity with [Livepeer Studio core concepts](/aboutstudio)
- Experience working with `cURL`
- Familiarity working with webhooks

## Technical Preparation

1. [A Livepeer account](https://livepeer.studio/login)

> 1. [An API key](https://livepeer.studio/docs/guides/start-live-streaming/api-key) (Bearer)

Reviewer: ```suggestion
1. [An API key](https://livepeer.studio/docs/guides/start-live-streaming/api-key)
```

1. [cURL](https://curl.se/)
1. Access to the [API Reference](https://livepeer.studio/docs/api-reference)

## **About Webhooks**

> Livepeer Studio uses webhooks to communicate with your application asynchronously when events for your stream occur. For example, you may want to know when a stream has become active or idle, so that you can surface this information to viewers. When these events happen, you can make a `POST` request to a URL that you specify.

Reviewer: ```suggestion
Livepeer Studio uses webhooks to communicate with your application asynchronously when events for your stream occur. For example, you may want to know when a stream has become `active` or `idle`, so that you can surface this information to viewers. When these events happen, you can make a `POST` request to a URL that you specify.
```

## Configuring Webhook Endpoints

Webhook endpoints are specific for each `event`. Once the event is registered, all streams for this account will be triggering the specific event. 

**For Example:** Registering for `stream.started` and `stream.idle` events:

### Request

> This cURL request registers a `stream.started` and `stream.idle` event (with a Bearer API key) so that all streams will trigger the `stream.started` and `stream.idle` events for this account.

Reviewer: ```suggestion
This cURL request registers a `stream.started` and `stream.idle` event so that all streams will trigger the `stream.started` and `stream.idle` events for this account.
```

  https://livepeer.studio/api/webhook
```

> ### Return

Reviewer: ```suggestion
### Response
```

> A JSON response with the events, event registration, `stream ID`, url of the webhook endpoint, and the `userID` of the stream

Reviewer: ```suggestion
Returns a JSON object with the `events`, stream `id`, `name` of the webhook, `url` of the webhook endpoint, and the `userID` of the stream
```

Reviewer: Removed "event registration" did not see that in the response object

| ------------------------ | ------------------------------------------------------------ |
| stream.started           | The parent stream object's isActive value is marked as true and the .m3u8 HLS URL works |
| stream.idle              | The parent stream object's isActive value should be marked as false and the .m3u8 HLS URL no longer works |
| recording.ready          | This fires when a recording is ready to be downloaded        |
| recording.started        | This fires when recording has started on an active stream    |
| recording.waiting        | This fires after a stream with recording on has concluded and is not yet ready to be downloaded. Typically it takes 5 minutes for recordings to be ready for download. |
| multistream.connected    | This fires when we've successfully connected to the multistream target |
| multistream.error        | This fires when we've encountered an error either while attempting to connect to the third party streaming service or while broadcasting. |
| multistream.disconnected | This fires when we are no longer sending video to the multistream target. |

## Retries

Currently, if a webhook fails, Livepeer will attempt to deliver the webhook for

> ~15 minutes with an exponential back off.

Reviewer: Move section to line 31, after "About Webhooks" section. Might be a good idea to introduce the different types of events to use before introducing on how to configure them.
